### PR TITLE
feat: add github ci workflow

### PR DESCRIPTION
**Pull request by fatelei**
currently, workflow only supports lint.

### CODE REVIEW

**Comment by fatelei** [comment]
@imeoer

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go
+      - name: Build
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.43.0
+          export PATH=$PATH:$(go env GOPATH)/bin
+          make check
```
Comment (bergwolf) [inline .github/workflows/ci.yml:32]: Is it possible to add a simple E2E CI?

Comment (fatelei) [reply .github/workflows/ci.yml:32]: i will try it

Comment (imeoer) [inline .github/workflows/ci.yml:32]: Maybe we can add `make` build further.

Comment (imeoer) [reply .github/workflows/ci.yml:32]: https://github.com/containers/nydus-storage-plugin/issues/8

Comment (fatelei) [reply .github/workflows/ci.yml:35]: image

done

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+GOOS ?= linux
+GOARCH ?= amd64
+
+.PHONY: build
+build:
+	GOOS=${GOOS} GOARCH=${GOARCH} ${PROXY} go build -ldflags="-s -w -X 'main.Version=${VERSION}'" -v -o bin/nydus-store ./cmd/store
```
Comment (imeoer) [inline Makefile:6]: Now we name the binary `nydus-store`, should also update the `README.md`.

File: .gitignore [hunk 3]
```diff
@@ -14,3 +14,4 @@
 # Dependency directories (remove the comment below to include it)
 # vendor/
 .idea
+.vscode
```
Comment (imeoer) [inline .gitignore:17]: Do we need to add `bin/nydus-store` here?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Log in to registry
+        # This is where you will update the PAT to GITHUB_TOKEN
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
+      - name: Build
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.48.0
+          export PATH=$PATH:$(go env GOPATH)/bin
+          make check
+          make build
+      - name: test
```
Comment (imeoer) [inline .github/workflows/ci.yml:37]: `Test`.

File: .github/workflows/nydus-config.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (imeoer) [inline .github/workflows/nydus-config.json:1]: Maybe move this file to `misc/nydus-config.json`, and update `README.md` to use this configuration. (it's not a blocker for this PR).

File: .github/workflows/storage.conf [hunk 6]
```diff
@@ -0,0 +1,236 @@
+# This file is is the configuration file for all tools
```
Comment (imeoer) [inline .github/workflows/storage.conf:1]: Maybe move this file to `misc/storage.conf`, and update `README.md` to use this configuration. (it's not a blocker for this PR).

**Review by imeoer — APPROVED** [review]
LGTM!

**Pull request merged**